### DISCUSSION TITLE
get_config with junos - save as type set/json?

### DISCUSSION THREAD

**Discussion by pointybirds** [napalm-automation/napalm#1403] [Ideas]
Hi all,

New Napalm user here, so bear with me: When using the get_config function for a junos device, it would be nice to be able to retrieve the config in the "set" or json style formats, rather than just the programmatic "text" version. As far as I can see with my limited skills, this is available in the "format" option of the options dict in the junos get_config function:

```
    def get_config(self, retrieve="all", full=False, sanitized=False):
        rv = {"startup": "", "running": "", "candidate": ""}

        options = {"format": "text", "database": "candidate"}
```

Does it seem reasonable to make this something the user can configure (with the default as text), or is there a better way to go about it? Apologies if this has been mentioned before, but I can't find anything about it.

**Comment by mirceaulinic** [comment]
Hi @pointybirds, yes, this make sense; there's also this FR: https://github.com/napalm-automation/napalm/issues/1105. If you'd like to work on this, feel free to go ahead and open a PR.
